zk-token-sdk/src/encryption/discrete_log.rs

```diff
@@ -141,35 +145,51 @@ impl DiscreteLog {
     /// Solves the discrete log problem under the assumption that the solution
     /// is a positive 32-bit number.
     pub fn decode_u32(self) -> Option<u64> {
-        let mut starting_point = self.target;
-        let handles = (0..self.num_threads)
-            .map(|i| {
-                let ristretto_iterator = RistrettoIterator::new(
-                    (starting_point, i as u64),
-                    (-(&self.step_point), self.num_threads as u64),
-                );
-
-                let handle = thread::spawn(move || {
-                    Self::decode_range(
-                        ristretto_iterator,
-                        self.range_bound,
-                        self.compression_batch_size,
-                    )
-                });
-
-                starting_point -= G;
-                handle
-            })
-            .collect::<Vec<_>>();
-
-        let mut solution = None;
-        for handle in handles {
-            let discrete_log = handle.join().unwrap();
-            if discrete_log.is_some() {
-                solution = discrete_log;
+        #[cfg(not(target_arch = "wasm32"))]
+        {
+            let mut starting_point = self.target;
+            let handles = (0..self.num_threads)
+                .map(|i| {
+                    let ristretto_iterator = RistrettoIterator::new(
+                        (starting_point, i as u64),
+                        (-(&self.step_point), self.num_threads as u64),
+                    );
+
+                    let handle = thread::spawn(move || {
+                        Self::decode_range(
+                            ristretto_iterator,
+                            self.range_bound,
+                            self.compression_batch_size,
+                        )
+                    });
+
+                    starting_point -= G;
+                    handle
+                })
+                .collect::<Vec<_>>();
+
+            let mut solution = None;
+            for handle in handles {
+                let discrete_log = handle.join().unwrap();
+                if discrete_log.is_some() {
+                    solution = discrete_log;
+                }
             }
+            solution
```

Comment:
nit: Sorry, this is my first time actually looking at this code. Is the idea that only one of the threads will return a result? If so, we can update this to:

```suggestion
        handles
            .into_iter()
            .map_while(|h| h.join().ok())
            .find(|x| x.is_some())
            .flatten()
```

But no worries if you don't like it or if it might not be correct

Reply:
Yes, much cleaner and easier to understand. Thanks!